aclk/aclk_query.c

```diff
@@ -198,17 +223,17 @@ aclk_query_handler aclk_query_handlers[] = {
 };
 
 
-static void aclk_query_process_msg(struct aclk_query_thread *info, aclk_query_t query)
+static void aclk_query_process_msg(aclk_query_t query)
 {
     for (int i = 0; aclk_query_handlers[i].type != UNKNOWN; i++) {
         if (aclk_query_handlers[i].type == query->type) {
             debug(D_ACLK, "Processing Queued Message of type: \"%s\"", aclk_query_handlers[i].name);
-            aclk_query_handlers[i].fnc(info->client, query);
+            aclk_query_handlers[i].fnc(aclk_query_thread->client, query);
```

Comment:
Here we pass just the ->client. Accessing thread local variable (with all of its possible performance hits).
Then in the function `http_api_v2` we do `aclk_query_thread->idx` accessing the thread local variable again, with performance hit again....
I havent finished thinking about it yet. But I start to think passing `aclk_query_thread` here instead of `aclk_query_thread->client` would be better. Then we can avoid also thread local variable (reasons for that are perf. and compatibility across compilers (what about MSVC?))